test.py

```diff
@@ -19,41 +19,236 @@
-    abs_script = os.path.abspath(sys.argv[0])
-
-    # The root directory of the script
-    abs_root = os.path.dirname(abs_script)
-
-    for file in os.listdir(os.path.join(abs_root, "example_configurations")):
-        if not file.endswith(".ini"):
-            continue
-        extra_options = ""
-        if args.debug:
-            extra_options = extra_options + " -d"
-        extra_options = extra_options + " -j" + str(args.j)
-        command = os.path.join(abs_root, "run.py") + " -l -t -c " + os.path.join(abs_root, "example_configurations", file) + " -o output_" + file + extra_options
-        print("Running " + command)
-        ret_program = subprocess.call(command, shell=True, executable="/bin/bash")
-        if ret_program:
-            print("Error in running " + file)
+    done_file_flag = os.path.join(args.output, 'done')
+
+    try:
+        os.mkdir(args.output)
+    except FileExistsError:
+        if os.path.exists(done_file_flag):
+            print(args.output+" already exists. Terminating the program...")
             sys.exit(1)
+        
+
+    tests = generate_tests() #TODO: documenta variabile tests (nel senso, controlla se/come documentarla)
+
+    #Perform tests
+    outcomes = []
+    for configuration in tests:
+        test_name = configuration.pop('Name')
+        output_path = args.output + '/' + test_name
+
+        try:
+            #Check if the test was already performed in a previous incomplete run
+            test_done_file_flag = os.path.join(output_path, 'done')
+            if os.path.exists(test_done_file_flag):
+                print(test_name,"already performed",sep=" ",end="\n\n\n")
+                outcomes.append(str(test_name)+" already performed")
+                continue
+            print("Starting",test_name,sep=" ")
+
+            if test_name == 'faas_predict':
+                # Build object
+                predictor_obj = Predictor(regressor_file=args.output+'/faas_test/LRRidge.pickle', output_folder=output_path, debug=args.debug)
+
+                # Perform prediction reading from a config file
+                predictor_obj.predict(config_file=configuration, mape_to_file=True)
+            else:
+                sequence_data_processor = sequence_data_processing.SequenceDataProcessing(configuration, debug=args.debug, output=output_path)
+                sequence_data_processor.process()
+        except Exception as e:
+            print("Exception",e,"raised", sep=' ')
+            outcomes.append(str(test_name)+" failed with exception "+str(e))
+        else:
+            outcomes.append(str(test_name)+" successfully run")
+        print('\n\n\n')
+
+    #Print results
+    print('\n\n\n\n\n\n\n\n\n\n\n\n*************Test Results*************')
+    i = 0
+    for outcome in outcomes:
+        i += 1
+        print(str(i)+')',outcome, sep=' ')
+
+    # Create success flag file
+    with open(done_file_flag, 'wb') as f:
+        pass
+
+
 
+def generate_tests():
```

Comment:
I would move this function before the `main()`.

Moreover, I would change the names of both `generate_test(s)` functions to be slightly more specific, so as not to create confusion.

Finally, a similar docstring as the other function would be useful. Here you could also specify all names of the performed tests